File: maximum-score-after-splitting-a-string.ts
{
    function maxScore(s) {
        let left = 0,
            right = s.split("").filter(x => x == '1').length,
            max = 0;

        for(let i = 0; i < s.length - 1; i++) {
            if(s[i] == '0') {
                left += 1
            } else {
                right -= 1
            }
            
            max = Math.max(max, left + right)
        }

        return max
    };

    console.log(maxScore('011101'))
}

File: maximum-score-after-splitting-a-string.rs
pub fn max_score(s: String) -> i32 {
    let mut left = 0;
    let mut right = s.chars().filter(|&x| x == '1').count() as i32;
    let mut max = 0;

    for i in 0..s.len() - 1 {
        if s.chars().nth(i).unwrap() == '0' {
            left += 1;
        } else {
            right -= 1;
        }
        max = max.max(left + right);
    }

    return max;
}